mariner/mariner.go

```diff
@@ -154,8 +152,24 @@ func (c Config) update(version, path string) error {
 
 	return nil
 }
+func (c Config) saveAdvisoryPerYear(dirName string, def Definition) error {
+	// Mariner uses `<ID>_<last_number_from_version>` format for `advisory_id`.
+	// But `advisory_id` is not required field.
+	// Therefore, if `advisory_id` is not exist, we create this field independently.
+	// cf. https://github.com/aquasecurity/vuln-list-update/pull/271#issuecomment-2111678641
+	advisoryID := def.Metadata.AdvisoryID
+	if advisoryID == "" {
+		advisoryID = def.ID
+		// for `0` versions `_0` suffix is omitted.
+		if def.Version != "" && def.Version[len(def.Version)-1:] != "0" {
+			advisoryID = fmt.Sprintf("%s_%s", advisoryID, def.Version[len(def.Version)-1:])
+		}
+	}
+	// Use advisory_id for file name to avoid overwriting files when there are 2 definitions for same CVE
+	// cf. https://github.com/aquasecurity/trivy-db/issues/379
+	fileName := fmt.Sprintf("%s.json", advisoryID)
 
-func (c Config) saveAdvisoryPerYear(dirName string, vulnID string, def Definition) error {
+	vulnID := def.Metadata.Reference.RefID
```

Comment:
I thought about using `advisory_date` field.
But this field does not always exist:
```
➜ cat cbl-mariner-1.0-oval.xml| grep ' <definition class="vulnerability"' | toilet -l
      2252
➜ cat cbl-mariner-1.0-oval.xml| grep '<advisory_date>' | toilet -l
      2070
```
So I'm leaving logic with year number from CVE.